app/src/main/java/ca/marshallasch/veil/services/VeilService.java

```diff
@@ -0,0 +1,143 @@
+package ca.marshallasch.veil.services;
+
+import android.app.Service;
+import android.content.Intent;
+import android.os.Handler;
+import android.os.HandlerThread;
+import android.os.IBinder;
+import android.os.Looper;
+import android.os.Message;
+import android.os.Messenger;
+import android.support.annotation.Nullable;
+import android.os.Process;
+import android.widget.Toast;
+
+import ca.marshallasch.veil.controllers.RightMeshController;
+
+/**
+ * Hosts all RightMesh logic on this service thread. Also receives {@link Message}s from
+ * {@link ca.marshallasch.veil.MainActivity} to handle work requested from application.
+ *
+ * @author Weihan Li
+ * @version 1.0
+ * @created on 2018-07-09
+ * @name veil-droid
+ */
+public class VeilService extends Service {
+
+    private RightMeshController rightMeshController;
+    private Looper veilServiceLooper;
+    private ServiceHandler veilServiceHandler;
+
+    //Message Strings
+    public static final int ACTION_VIEW_MESH_SETTINGS = 1;
+    public static final int ACTION_MAIN_RESUME_MESH = 2;
+    public static final int ACTION_MAIN_REFRESH_PEER_LIST = 3;
+    public static final int ACTION_MAIN_MANUAL_REFRESH = 4;
+
+
+    /**
+     * ServiceHandler class for the {@link VeilService}
+     */
+    private final class ServiceHandler extends Handler {
+        public ServiceHandler() {
```

Comment:
whats this constructor for it does not do anything?

Reply:
Removed. Sorry left that in when I was following guide. Nice catch.